src/gui-qt4/libs/seiscomp3/gui/core/connectiondialog.cpp

```diff
@@ -217,7 +217,7 @@ bool ConnectionDialog::fetchDatabase() {
 		Util::StopWatch timer;
 		while ( timer.elapsed() < TimeSpan(5.0) ) {
 			(*_connection)->poll();
-			while ( msg = (*_connection)->readMessage(false) ) {
+			while ( msg == (*_connection)->readMessage(false) ) {
```

Comment:
Yes, the warning is right. We have to enclose the condition with parenthesis. I will do that in master because it is a general issue.